components/expenses/PayoutBankInformationForm.js

```diff
@@ -0,0 +1,243 @@
+import { useQuery } from 'react-apollo';
+import { Box, Flex } from '@rebass/grid';
+import { get } from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field, FastField, useFormikContext } from 'formik';
+import { defineMessages, useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import StyledInputField from '../StyledInputField';
+import StyledInput from '../StyledInput';
+import StyledSpinner from '../StyledSpinner';
+import StyledSelect from '../StyledSelect';
+import { P } from '../Text';
+
+import { formatErrorMessage } from '../../lib/form-utils';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { PayoutMethodType } from '../../lib/constants/payout-method';
+
+const formatStringOptions = strings => strings.map(s => ({ label: s, value: s }));
+const formatTransferWiseSelectOptions = values => values.map(({ key, name }) => ({ label: name, value: key }));
+
+export const msg = defineMessages({
+  currency: {
+    id: 'Currency',
+    defaultMessage: 'Currency',
+  },
+});
+
+const accountHolderFieldOptions = {
+  name: 'Account Holder Name',
+  group: [
+    {
+      required: true,
+      key: 'accountHolderName',
+      name: 'Account Holder Name',
+      type: 'text',
+      example: 'Jane Doe',
+    },
+  ],
+};
+
+const FormFieldsContainer = styled(Flex)`
+  text-transform: capitalize;
+`;
+
+const requiredFieldsQuery = gqlV2`
+  query Collective($slug: String, $currency: TransferWiseCurrency!) {
+    collective(slug: $slug) {
+      host {
+        transferwise {
+          requiredFields(currency: $currency) {
+            type
+            title
+            fields {
+              name
+              group {
+                key
+                name
+                type
+                required
+                example
+                valuesAllowed {
+                  key
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+const RequiredFields = ({ disabled, getFieldName, formik, collective, currency }) => {
+  const { loading, error, data } = useQuery(requiredFieldsQuery, {
+    context: API_V2_CONTEXT,
+    variables: { slug: collective.slug, currency },
+  });
+  React.useEffect(() => {
+    const type = get(data, 'collective.host.transferwise.requiredFields[0].type');
+    if (type) {
+      formik.setFieldValue(getFieldName('data.type'), type);
+    }
+  }, [data]);
+
+  if (loading) {
+    return <StyledSpinner />;
+  }
+  if (error) {
+    return <P>{error.message}</P>;
+  }
+
+  const { fields, title } = data.collective.host.transferwise.requiredFields[0];
```

Comment:
It's valid to submit expenses to collectives without hosts, `data.collective.host` can be null

Reply:
But in that case this Form wouldn't be rendered, right?

Reply:
Ah yes, correct :slightly_smiling_face: